pkg/scheduler/scheduler.go

```diff
@@ -114,10 +118,14 @@ func (s *scheduler) Filter(args *schedulerapiv1.ExtenderArgs) (*schedulerapiv1.E
 	for _, predicate := range predicatesByComponent {
 		glog.Infof("entering predicate: %s, nodes: %v", predicate.Name(), predicates.GetNodeNames(kubeNodes))
 		kubeNodes, err = predicate.Filter(instanceName, pod, kubeNodes)
+		glog.Infof("leaving predicate: %s, nodes: %v", predicate.Name(), predicates.GetNodeNames(kubeNodes))
 		if err != nil {
-			return nil, err
+			s.recorder.Event(pod, apiv1.EventTypeWarning, predicate.Name(), err.Error())
+			if len(kubeNodes) == 0 {
+				// do not return error to k8s: https://github.com/pingcap/tidb-operator/issues/1353
+				return nil, nil
+			}
```

Comment:
Does `nil` semantically equal to filter result with empty nodes? If it is, these three lines seem unnecessary because, at the end of this function, it will return filter results with an empty node list.

Reply:
> Does nil semantically equal to filter result with empty nodes? 

Yes.

> If it is, these three lines seem unnecessary because, at the end of this function, it will return filter results with an empty node list.

This `return` is in a `for range` loop. No need to step the next predicate when the `kubeNodes` is empty.

Reply:
What about to change `return nil, nil` to break?

```
if len(kubeNodes) == 0 {
	break
}
```

Reply:
I think it's good

Reply:
fixed PTAL